app/services/remote_identities/create_service.rb

```diff
@@ -45,8 +45,11 @@ def initialize(user:, integration:, token:)
     end
 
     def call
-      @model.origin_user_id = @integration.extract_origin_user_id(@token)
-      if @model.changed?
+      if @model.new_record?
```

Comment:
:red_circle: 
I am not sure about that change. At least the following case is possible which I consider incorrect:
1. OP user has OAuth connection to NC(OAuthClientToken and RemoteIdentity)
2. Then OP user removes OAuthClientToken through UI, RemoteIdentity remains.
3. User goes though OAuth grant flow, but logs in as a different user on NC side.
4. Exisiting RemoteIdentity is not updated and contains wrong origin_user_id.

Reply:
Agreed. That case is a problem... Let me think about it.

Reply:
Here's my idea, tell me what you think:

* `RemoteIdentities::CreateService` accepts a flag that forces an update of the remote identity
* ConnectionManager sets that flag, SSO doesn't

In general I think all of this is just a workaround. A really nice thing would be to offer the user management/transparency about remote identities, the same as we show for access tokens for other applications. But as long as we don't have that, this approach could solve most of the issues we see:

* the case you described does not occur anymore
* we do not flood the storage with queries about the user identity from SSO; the OAuth 2.0 case makes sure that we don't do that anyways, by not calling the `CreateService` every time

Reply:
I'm fine with this idea.

Reply:
implemented it